### PR TITLE
Make Spiky portable when frozen

### PR DESCRIPTION
**Pull request by weluvgoatz**
Hey,  I learned some C++ (thanks @Nobelix :))  and decided to give this another shot. Please tell me if it works or not, and if it doesn't, please explain. Thanks.

### CODE REVIEW

File: spiky.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
+//  SuperTux
```
Comment (tobbi) [inline spiky.cpp]: No need for this file and the .hpp file

File: src/badguy/spiky.hpp [hunk 2]
```diff
@@ -18,13 +18,16 @@
 #define HEADER_SUPERTUX_BADGUY_SPIKY_HPP
 
 #include "badguy/walking_badguy.hpp"
+#include "object/portable.hpp"
 
 class Spiky : public WalkingBadguy
+              public Portable
```
Comment (qwertychouskie) [inline src/badguy/spiky.hpp]: This causes a compilation error:
```
In file included from /home/travis/build/SuperTux/supertux/src/badguy/spiky.cpp:17:0:
/home/travis/build/SuperTux/supertux/src/badguy/spiky.hpp:24:15: error: expected ‘{’ before ‘public’
               public Portable
               ^
/home/travis/build/SuperTux/supertux/src/badguy/spiky.hpp:24:15: error: expected unqualified-id before ‘public’
make[2]: *** [CMakeFiles/supertux2_lib.dir/src/badguy/spiky.cpp.o] Error 1
make[1]: *** [CMakeFiles/supertux2_lib.dir/all] Error 2
make: *** [all] Error 2
The command "make install DESTDIR="/tmp/supertux-$BUILD_TYPE"" exited with 2.
```

Comment (christ2go) [reply src/badguy/spiky.hpp]: There is a `,` missing.

**Comment by weluvgoatz** [comment]
does it work? Is it better?

**Comment by tobbi** [comment]
Why did you close this?

**Comment by weluvgoatz** [comment]
Oh, closed because it seems this is already being worked on in Issue #683 .

**Pull request closed without merging**